src/util/uuid/mod.rs

```diff
@@ -0,0 +1,110 @@
+use std::char;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub fn uuid() -> String {
+    let mut numbers = [0u8; 36];
+    make_random_numbers(&mut numbers);
+    uuid_from_numbers(&numbers)
+}
+
+#[cfg(target_arch = "wasm32")]
+fn make_random_numbers(numbers: &mut [u8]) {
+    web_sys::window()
+        .expect("window is not available")
+        .crypto()
+        .expect("window.crypto is not available")
+        .get_random_values_with_u8_array(numbers)
+        .expect("window.crypto.getRandomValues not available");
```

Comment:
better to return an error?

Reply:
In principle yes, but changing this turned into a big yak shave in embed, so I'm putting it off for now.